src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,108 @@
+package duke;
+
+import java.time.LocalDate;
+import java.util.Arrays;
+import java.util.List;
+
+import duke.command.AddCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.MarkCommand;
+import duke.command.UnknownCommand;
+import duke.command.UnmarkCommand;
+import duke.exception.DukeException;
+import duke.exception.EmptyDescriptionException;
+import duke.exception.MissingIndexException;
+import duke.exception.UnknownCommandException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Todo;
+
+
+/**
+ * The duke.Parser class takes an input and returns
+ * a corresponding command.
+ *
+ * @author Zi Xiang
+ * @version CS2103 AY23/24 Sem 1
+ */
+public class Parser {
+    // Initialisation of objects and variables
+    private static List<String> commands = Arrays.asList(
+            new String[]{"todo", "deadline", "event", "mark", "unmark", "delete", "find"});
+
+    /**
+     * Returns a corresponding Command from input.
+     *
+     * @param fullCommand a String with the command
+     * @return A Command to be executed
+     * @throws DukeException If input does not match
+     */
+    public static Command parse(String fullCommand) throws DukeException {
```

Comment:
Maybe you could try incorporating the Switch/Case statement for this method if it is possible for better readability? 🤔

Reply:
Ok sure, will try to make that change